Tests/Realm.Tests/Database/NotificationTests.cs

```diff
@@ -195,6 +195,179 @@ public void DictionaryNotificationToken_KeepsCollectionAlive()
             });
         }
 
+        [Test]
+        public void PrimitivePropertyInObjectShouldFireNotificationOnChange()
+        {
+            var testObject = new TestNotificationObject();
+            _realm.Write(() => _realm.Add(testObject));
+            var notificationCount = 0;
+            testObject.PropertyChanged += (sender, e) =>
+            {
+                notificationCount++;
+            };
+            _realm.Write(() =>
+            {
+                testObject.StringProperty = "foo";
+            });
+            _realm.Refresh();
+            Assert.That(notificationCount, Is.EqualTo(1));
+        }
+
+        [Test]
+        public void CollectionPropertiesOfDifferentTypeShouldNotFireNotificationsOnChange()
+        {
+            var testObject = _realm.Write(() => _realm.Add(new TestNotificationObject()));
+            testObject.PropertyChanged += (sender, e) =>
+            {
+                Assert.That(e.PropertyName, Is.EqualTo("StringProperty"));
```

Comment:
I don't think this is the most correct approach. For example, the test will pass even if there will be a notification for StringProperty when you modify something else. 
You probably should create a list of strings and just add the `PropertyName` to the list in `PropertyChanged` after every `_realm.Refresh()` you can check if the list contains the elements you expect. Finally, you clear the list so it's empty for next assertion.